extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -324,12 +324,23 @@ public function getPage() {
 				'ns' => $title->getNamespace()
 			];
 
-			if ( $this->isSupportedByMercury( $title ) ) {
-				// Empty category pages are not known but contain article list
+			// handle cases like starwars.wikia.com/wiki/w:c:clashroyale:Tesla (interwiki links)
+			$interWikiUrl = '';
+			InterwikiDispatcher::getInterWikiaURL( $title, $interWikiUrl, '' );
+
+			if ( empty( $interWikiUrl ) && $this->isSupportedByMercury( $title ) ) {
+				// Empty category pages are not known but contain article list;
 				if ( !$title->isKnown() && $title->getNamespace() !== NS_CATEGORY ) {
 					throw new NotFoundApiException( 'Page doesn\'t exist' );
 				}
 
+				// InterwikiDispatcher::getInterWikiaURL does not support other prefixes than InterwikiDispatcher::SUPPORTED_IW_PREFIXES
+				// but other prefixes may be defined in `interwiki` table for given wiki - in such case $title->isKnown()
+				// returns true in previous `if` statement
+				if ( !empty( $title->mInterwiki ) && !in_array($title->mInterwiki, InterwikiDispatcher::SUPPORTED_IW_PREFIXES ) ) {
```

Comment:
What do you think about InterwikiDispatcher::isSupportedPrefix($prefix)?
and adding a test for it?